File: src/main/java/econtroller/sensor/SensorChannel.java
package econtroller.sensor;

import cloud.elb.SenseData;
import econtroller.controller.Sense;
import econtroller.controller.StartSense;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-04-11
 *
 */

public class SensorChannel extends PortType {
	{
		negative(Sense.class);
		negative(StartSense.class);
		negative(StopSense.class);
		positive(SenseData.class);
	}
}

File: src/main/java/econtroller/sensor/SensorChannel.scala
package econtroller.sensor

import se.sics.kompics.PortType
import econtroller.controller.{StartSense, Sense}
import cloud.elb.SenseData

/**
 * @author Amir Moulavi
 * @date 2011-07-23
 *
 */
class SensorChannel extends PortType {

  negative(classOf[Sense])
  negative(classOf[StartSense])
  negative(classOf[StopSense])
  positive(classOf[SenseData])

}